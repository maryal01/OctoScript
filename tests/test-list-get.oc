list<int> l = [10,20,30,40];
int i = 0;
while(i < l.length()){
    int elem = l.get(i);
    print("(%d, %d), ", i, elem);
    i = i + 1;
}

list<float> f = [3.14, 6.28, 2419.948, 84.000011, 0.0];
i = 0;
while(i < f.length()){
    float elem = f.get(i);
    print("(%d, %f), ", i, elem);
    i = i + 1;
}

list<string> s = ["Sphinx", "of", "black", "quartz", "judge", "my", "vow"];
i = 0;
while(i < s.length()){
    string elem = s.get(i);
    print("(%d, %s), ", i, elem);
    i = i + 1;
}

list<bool> b = [true, false, true, true];
i = 0;
while(i < b.length()){
    bool elem = b.get(i);
    print("(%d, %d), ", i, elem);
    i = i + 1;
}
