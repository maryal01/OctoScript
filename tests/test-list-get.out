(0, 10), (1, 20), (2, 30), (3, 40), (0, 3.140000), (1, 6.280000), (2, 2419.948000), (3, 84.000011), (4, 0.000000), (0, Sphinx), (1, of), (2, black), (3, quartz), (4, judge), (5, my), (6, vow), (0, 1), (1, 0), (2, 1), (3, 1), 